ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/sync/worldstate/CompleteTaskStep.java

```diff
@@ -59,6 +76,23 @@ public void markAsCompleteOrFailed(
     }
   }
 
+  private void displayWorldStateSyncProgress() {
+    final String lastEstimatedWorldStateCompletion =
+        doubleFormatter.format(estimatedWorldStateCompletion);
+    final String newEstimatedWorldStateCompletion =
+        doubleFormatter.format(computeWorldStateSyncProgress());
+    if (!Objects.equals(newEstimatedWorldStateCompletion, lastEstimatedWorldStateCompletion)) {
+      LOG.info("Estimated World State completion: {}", newEstimatedWorldStateCompletion);
```

Comment:
As mentioned the percent complete winds up being misleading.  This should probably be something like:
`Downloaded {} world state nodes. At least {} nodes remaining.`. That conveniently avoids most of the complexity too. :)

Reply:
Done.

Reply:
I suggest keeping the percent as an estimation at the end of the log because i think it was the initial request.